screen-search/src/main/java/com/ivy/search/SearchScreen.kt

```diff
@@ -34,43 +28,32 @@ import com.ivy.legacy.utils.keyboardOnlyWindowInsets
 import com.ivy.legacy.utils.keyboardVisibleState
 import com.ivy.legacy.utils.onScreenStart
 import com.ivy.legacy.utils.selectEndTextFieldValue
+import com.ivy.navigation.IvyPreview
 import com.ivy.navigation.SearchScreen
+import com.ivy.navigation.screenScopedViewModel
 import com.ivy.resources.R
 import com.ivy.wallet.ui.theme.modal.DURATION_MODAL_ANIM
-import kotlinx.collections.immutable.ImmutableList
 import kotlinx.collections.immutable.persistentListOf
 
 @Composable
 fun SearchScreen(screen: SearchScreen) {
-    val viewModel: SearchViewModel = viewModel()
-
-    val transactions by viewModel.transactions.collectAsState()
-    val baseCurrency by viewModel.baseCurrencyCode.collectAsState()
-    val categories by viewModel.categories.collectAsState()
-    val accounts by viewModel.accounts.collectAsState()
+    val viewModel: SearchViewModel = screenScopedViewModel()
+    val uiState = viewModel.uiState()
 
     onScreenStart {
-        viewModel.search("")
+        viewModel.onEvent(SearchEvent.Search(""))
```

Comment:
Put this as `LaunchedEffect(Unit)` in the ViewModel's `@Composable fun uiState()`.
This way it'll still lauch on screen start but won't use legacy code and wlll be entirely controlled by the VM.